searx/engines/niconico.py

```diff
@@ -0,0 +1,85 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+"""Niconico search engine for searxng"""
+
+import time
+
+from urllib.parse import urlencode
+from datetime import datetime, timedelta
+from lxml import html
+
+from searx.utils import extract_text
+
+about = {
+    "website": "https://www.nicovideo.jp/",
+    "wikidata_id": "Q697233",
+    "use_official_api": False,
+    "require_api_key": False,
+    "results": "HTML",
+    "language": "ja",
+}
+
+categories = ["videos"]
+paging = True
+
+time_range_support = True
+time_range_dict = {"day": 86400, "week": 604800, "month": 2592000, "year": 31536000}
+
+base_url = "https://www.nicovideo.jp"
+
+
+def request(query, params):
+    query_params = {"page": params['pageno']}
+
+    if time_range_dict.get(params['time_range']):
+        now = int(time.time())
+        past = now - time_range_dict[params["time_range"]]
```

Comment:
I think using `datetime` and `timedelta` would be cleaner here than calculating with integers, see https://github.com/searxng/searxng/pull/4509/files#diff-704c60f5e4ecffba5ee315ed9698f013f905b5923d232a313632dee21e5df528R46 for example.